src/CoreWCF.Primitives/src/CoreWCF/Channels/DelegatingMessage.cs

```diff
@@ -0,0 +1,90 @@
+using System.Xml;
+using CoreWCF.Channels;
+using CoreWCF;
+
+namespace CoreWCF.Channels
+{
+
+    abstract class DelegatingMessage : Message
+    {
```

Comment:
Nit: extra blank line
Also needs accessor (internal) as does the innerMessage field.